src/main/java/duke/Commands/AddInput.java

```diff
@@ -0,0 +1,106 @@
+                }
+                if(Parser.isList(inputList[0].trim().toLowerCase())) {
+                    StringBuffer result = new StringBuffer();
+                    //to add in the starting line of the section
+                    result.append("Here are the tasks in your list:\n");
+
+                    for (int i = 0; i < taskList.getSize(); i++) {
+                        // getting the current task
+                        Task currentTask = taskList.getTask(i);
+
+                        // adding the current task into the taskList
+                        result.append((i + 1) + ". " + currentTask.toString() + "\n");
+                    }
+                    Ui.lineFormatter(result.toString());
+                }
+
+                //case where the command is incomplete, in the cases of done, todo, event, deadline and delete
+                else if(inputList.length < 2) {
+                    throw new IncompleteCommandException();
+
+                }
+                //Checks for FIND_KEY, to find list of tasks containing the keyword
+                else if(Parser.isFind(inputList[0].trim().toLowerCase())){
+                    Find.find(inputList[1].trim(), taskList);
+                }
+
+                // case where the input is done
+                else if(Parser.isComplete(inputList[0].trim().toLowerCase()) && Parser.isNum(inputList[1])){
+                    int currentIndex = Integer.parseInt(inputList[1]) - 1;
+                    if(currentIndex + 1> taskList.getSize() || currentIndex + 1 <= 0){
+                        throw new DoneException(currentIndex, taskList.getSize());
+                    } else {
+                        Task task = taskList.getTask(currentIndex);
+                        // to check if the task is already done
+                        if(task.getStatus()){
+                            throw new TaskAlreadyDoneException(task);
+//                            lineFormatter("This task is already done!\n" +
+//                                    task.toString());
+                            // if task is not done
+                        } else {
+                            taskList.getTask(currentIndex).markAsDone();
+                            Ui.taskDone(taskList.getTask(currentIndex));
+                        }
+                    }
+                } else if(Parser.isDelete(inputList[0].trim().toLowerCase()) && Parser.isNum(inputList[1])) {
+                    int currentIndex = Integer.parseInt(inputList[1]) - 1;
+                    if (currentIndex + 1 > taskList.getSize() || currentIndex + 1 <= 0) {
+                        throw new DeleteException(currentIndex, taskList.getSize());
+                    } else {
+                        Task deletedTask = taskList.removeTask(currentIndex);
+                        Ui.taskDeleted(deletedTask);
+                    }
+                } else {
+                    AddToList.addedToList(taskList, input);
+
+
```

Comment:
You may want to consider reducing the length of this function and also it may be a good idea to not nest conditional stmts inside the condition as there are already many conditional stmts. Blocks of code that are commented out can be removed.